src/tile.hpp

```diff
@@ -72,6 +74,22 @@ struct TileId
   }
 };
 
+struct TileMapInformation
+{
+  int zoom;
+
+  // local map information
+  bool local_map;
+  double meter_per_pixel_z0;
+  double origin_x;
+  double origin_y;
+  std::string origin_crs;
+
+  // local map projection
+  PJ_CONTEXT * context = proj_context_create();
```

Comment:
Thanks for this PR!

Would you mind using smart pointers here?

Reply:
Thank you for your feedback on this pull request.

You could use smart pointers (potentially `std::unique_ptr `), but objects of type `TileMapInformation` are passed into different functions. Thus, an explicit copy constructor would have to be defined for `TileMapInformation`. In addition, you would probably also define a custom destructor that calls special "cleanup" functions of the `PROJ` Lib. 

Since the use of smart pointers has been largely avoided in the repository, I have not used them as well. But of course, we can discuss this again. What do you think?